### ISSUE TITLE
How to make the network smaller?

### ISSUE THREAD

**Issue by dusa2** [open-mmlab/mmskeleton#281]
I managed to get 67% accuracy with the 2D dataset I have - that have less joints, almost always two skeletons in the frames, and video segments of 90 frames (3 seconds). Although the drawbacks on skeleton 2D dataset is addressed in the paper, I believe that for my dataset, this approach might have tremendous potential and I really want to make it work!

My problem is that I can not decrease val loss after some epochs, although the training loss keeps going down neatly. So it overfits.

Here are things I tried to decrease overfitting but did not improve the accuracy:
using adam optimizer
using adam optimizer with a scheduler that changes learning rate when there is a plateu
**data augmentation** ( random flip, crop, shear, scale, rotation) in addition to st-gcn's -> Somehow data augmentation made things even **worse**, I also tried original data+augmented data and it performed much worse
I tried a bunch of variables for learning rates and steps but no luck
I can not use transfer learning to use your pretrained model to start my network because I have less joints.

I am thinking maybe reducing network size/ layers could work, so I checked the net and st-gcn parts of the code but, I am not sure how I can do this. Any advice?

edit: Would decreasimg the number of units below work? Also is the kernel size set at 9, related to the number of frames (temporal dimension)?

>   self.conv0 = nn.Conv1d(channel, 64, kernel_size=9, padding=4)
>         conv_init(self.conv0)
> 
>         self.unit1 = TCN_unit(64, 64)
>         self.unit2 = TCN_unit(64, 64)
>         self.unit3 = TCN_unit(64, 64)
>         self.unit4 = TCN_unit(64, 128, stride=2)
>         self.unit5 = TCN_unit(128, 128)
>         self.unit6 = TCN_unit(128, 128)
>         self.unit7 = TCN_unit(128, 256, stride=2)
>         self.unit8 = TCN_unit(256, 256)
>         self.unit9 = TCN_unit(256, 256)
>         self.bn = nn.BatchNorm1d(256)
>         self.relu = nn.ReLU()
> 
>         self.fcn = nn.Conv1d(256, num_class, kernel_size=1)
> 
>     def forward(self, x):
>         N, C, T, V, M = x.size()
>         x = x.permute(0, 4, 3, 1, 2).contiguous().view(N, M * V * C, T)
> 
>         if self.use_data_bn:
>             x = self.data_bn(x)
> 
>         x = self.conv0(x)
>         x = self.unit1(x)
>         x = self.unit2(x)
>         x = self.unit3(x)
>         x = self.unit4(x)
>         x = self.unit5(x)
>         x = self.unit6(x)
>         x = self.unit7(x)
>         x = self.unit8(x)
>         x = self.unit9(x)
>         x = self.bn(x)
>         x = self.relu(x)

**Comment by yjxiong** [comment]
This question is not an issue of the codebase and we do not provide network tuning service.

Closing this. Please feel free to reopen it if you meet any further problem.
